todo:


*   Quitar columnas no deseadas
*   Limpiar/transformar datos faltantes
*   Pasar a numérico la variables categóricas
*   Hacer estadísticos de comportamiento, ya sea por su año de egreso o por la carrera a la que pertenecen.






NOTA: No hay variables numéricas como tal, sino categóricas.

In [1]:
# https://docs.google.com/forms/d/e/1FAIpQLSfLhw6dS0ASc1UUd2_xHCPoBgNlCqP9dF8q1EJEBR8Lv1KVSQ/viewform
# Suprimir advertencias:
def warn(*args, **kwargs):
    pass
import warnings
warnings.warn = warn

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.cluster import KMeans
from sklearn.preprocessing import StandardScaler

In [3]:
pathVS = 'C:\\Users\\ivans\\OneDrive\\Documents\\Escuela\\Semestre 2023-2\\Fundamentos de Ciencias de los Datos\\PROYECTO FINAL\\CUESTIONARIO DE SEGUIMIENTO DE EGRESADOS (respuestas).xlsx'
pathCOLL = 'CUESTIONARIO DE SEGUIMIENTO DE EGRESADOS (respuestas).xlsx'

In [5]:
data = pd.read_excel(pathCOLL)
data.head()

Marca temporal Carrera profesional cursada  \
0 2023-03-07 16:28:18.869            Ing. Electrónica   
1 2023-03-10 16:57:54.679            Ing. Electrónica   
2 2023-03-10 18:05:01.889            Ing. Electrónica   
3 2023-03-15 10:28:58.280            Ing. Electrónica   
4 2023-03-15 13:29:41.387              Ing. Biomédica   

             Lugar dónde reside Titulado  Dominio del idioma Inglés  \
0          Hermosillo, Son, Mex       No                        0.8   
1   Hermosillo, Sonora, México        Si                        1.0   
2     Hermosillo,Sonora,Mexico        Si                        0.6   
3    Hermosillo Sonora, Mexico        Si                        0.8   
4  Guadalajara, Jalisco, Mexico       Si                        0.8   

  Indica semestre y año de ingreso  Indica semestre y año de egreso  \
0               2016-08-22 00:00:00             2020-12-18 00:00:00   
1               2015-08-10 00:00:00             2019-12-02 00:00:00   
2               2017-01-31 00:00:00             2020-06-20 00:00:00   
3               2016-08-30 00:00:00             2021-01-20 00:00:00   
4               2017-08-20 00:00:00             2022-01-10 00:00:00   

  ¿Tu egreso fue hace más de 2 años? Calidad de los docentes Plan de estudios  \
0                                 Si                     NaN              NaN   
1                                 Si                     NaN              NaN   
2                                 Si                     NaN              NaN   
3                                 Si                     NaN              NaN   
4                                 No                   Buena          Regular   

   ...  \
0  ...   
1  ...   
2  ...   
3  ...   
4  ...   

  Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Pensamiento critico]  \
0                                      Si la utilizo                                                                                                               
1                                      Si la utilizo                                                                                                               
2                                      Si la utilizo                                                                                                               
3                                      Si la utilizo                                                                                                               
4                                                NaN                                                                                                               

  Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Adaptación]  \
0                                      Si la utilizo                                                                                                      
1                                      Si la utilizo                                                                                                      
2                                      Si la utilizo                                                                                                      
3                                      Si la utilizo                                                                                                      
4                                                NaN                                                                                                      

  Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Innovación]  \
0                                      Si la utilizo                                                                                                      
1                                      S

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 695 entries, 0 to 694
Data columns (total 83 columns):
 #   Column                                                                                                                                                                                                                                                                                                                                                                                             Non-Null Count  Dtype         
---  ------                                                                                                                                                                                                                                                                                                                                                                                             --------------  -----         
 0   Marca temporal                              

<h1>
1. LIMPIEZA
<h1/>

In [7]:
dataRecortado = data.drop(['Marca temporal', 'Nivel jerárquico de tu empleo', 'Nombre de la empresa u organización donde trabaja actualmente', 'Si en la pregunta anterior indicaste alguna opción que incluía la palabra "OTRO", favor de indicar el tipo de empresa; en caso de NO haber seleccionado "OTRO", responder NA', 'Si tu respuesta es afirmativa, indique en cuáles cursos está interesado.',
           'Si tu respuesta anterior fue afirmativa, indica en qué posgrado estás interesado(a):', 'Si seleccionaste la opción "OTRO", favor de especificar el posgrado y la institución donde estudiará:', 'Actividad a la que se dedica actualmente', 'Si estudia, indicar si es', 'Nombre de estudios que realiza actualmente', 'Nombre de Institución donde estudia actualmente',
           'Nombre de la compañía donde trabaja actualmente', 'Datos de la empresa u organismo.                 ', 'Sector económico de la empresa u organización donde trabaja.1', 'Si en la pregunta anterior indicaste alguna opción que incluía la palabra "OTRO", favor de indicar el tipo de empresa; en caso de NO haber seleccionado "OTRO" responder NA',
           'Tamaño de la empresa u organización (número de empleados en la empresa).1', 'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Liderazgo]', 'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Motivación]',
           'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Trabajo en equipo]', 'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Comunicación]', 'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Organización]',
           'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Pensamiento critico]', 'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Adaptación]', 'Selecciona según sea el caso, cuáles de las siguientes habilidades son las que más utilizas en el desarrollo de su actividad profesional [Innovación]',
           'Si su respuesta es afirmativa, indique a qué organización social pertenece', 'Si su respuesta es afirmativa, indique a qué organismo de profesionistas pertenece\n\n', 'Si su respuesta es afirmativa, indique a qué asociación de egresados pertenece', 'Opinión o recomendación para mejorar la formación profesional de un egresado de su carrera'], axis=1)

In [8]:
dataRecortado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 695 entries, 0 to 694
Data columns (total 55 columns):
 #   Column                                                                                                                                                                                                                                                                                                                                                                                             Non-Null Count  Dtype  
---  ------                                                                                                                                                                                                                                                                                                                                                                                             --------------  -----  
 0   Carrera profesional cursada                               

In [ ]:
path2Save = 'C:\\Users\\ivans\\OneDrive\\Documents\\Escuela\\Semestre 2023-2\\Fundamentos de Ciencias de los Datos\\PROYECTO FINAL\\Primera_Revision_Columnas_Limpiadas_Encuesta.xlsx'

In [ ]:
path2Save = 'C:\\Users\\ivans\\OneDrive\\Documents\\Escuela\\Semestre 2023-2\\Fundamentos de Ciencias de los Datos\\PROYECTO FINAL\\Primera_Revision_Columnas_Limpiadas_Encuesta.xlsx'
dataRecortado.to_excel(path2Save, index = False)

## De 83 a 55 columnas

In [17]:
dataRecortado = dataRecortado.rename(columns = lambda x: x.strip())
print("\nColumnas después del cambio:")
print(dataRecortado.columns)


Columnas después del cambio:
Index(['Carrera profesional cursada', 'Lugar dónde reside', 'Titulado',
       'Dominio del idioma Inglés', 'Indica semestre y año de ingreso',
       'Indica semestre y año de egreso', '¿Tu egreso fue hace más de 2 años?',
       'Calidad de los docentes', 'Plan de estudios',
       'Oportunidad de participar en proyectos de investigación y desarrollo',
       'Énfasis que se le prestaba a la investigación dentro del proceso de enseñanza',
       'Satisfacción con las condiciones de estudio (infraestructura)',
       'Experiencia obtenida a través de la residencia profesional / práctica profesional',
       '¿Conseguiste trabajo en la empresa donde realizaste tus residencias profesionales?',
       'Condición de trabajo',
       'Indica la relación del trabajo con tu área de formación profesional:',
       'Datos de la empresa u organismo',
       'Sector económico de la empresa u organización donde trabaja',
       'Tamaño de la empresa u organización (n

In [18]:
dataRecortado.isnull().sum()

Carrera profesional cursada                                                                                                                                                                                                                                                                                                                                                                            0
Lugar dónde reside                                                                                                                                                                                                                                                                                                                                                                                     1
Titulado                                                                                                                                                                                                              

In [19]:
dataRecortado['Carrera profesional cursada'].count()

695

In [ ]:
pathCOLL = 'Primera_Revision_Columnas_Limpiadas_Encuesta.xlsx'

In [ ]:
data = pd.read_excel(pathCOLL)

Cantidad de datos faltantes por las columnas más graves.

In [20]:
data['Selecciona según sea el caso, cuál de las siguientes habilidades son las que más utilizas en el desarrollo de tu actividad profesional: [Innovación]'].isnull().sum() * 100 / data['Carrera profesional cursada'].count()

56.115107913669064

In [21]:
for column in data.columns:
  faltante = data[column].isnull().sum() * 100 / data['Carrera profesional cursada'].count()
  print(faltante)

0.0
0.0
0.14388489208633093
0.0
0.0
0.0
0.0
0.0
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
61.15107913669065
56.115107913669064
87.62589928057554
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
56.115107913669064
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
38.56115107913669
0.0
0.0
84.1726618705036
43.884892086330936
43.884892086330936
81.29496402877697
82.01438848920863
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
48.63309352517986
43.884892086330936
43.884892086330936
82.01438848920863
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.8

En total son 695 datos, existen columnas más del 50% de los datos faltantes. Para evitar alterar la integridad de los datos de esas columnas a la hora de aplicar promedios y/o modas, se descartarán aquellas columnas que tengan más del 50% de los datos faltantes.

In [22]:
for column in data.columns:
  faltante = data[column].isnull().sum() * 100 / data['Carrera profesional cursada'].count()
  if faltante > 50:
    data = data.drop([column], axis=1)

In [23]:
for column in data.columns:
  faltante = data[column].isnull().sum() * 100 / data['Carrera profesional cursada'].count()
  print(faltante)

0.0
0.0
0.14388489208633093
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
38.56115107913669
0.0
0.0
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
48.63309352517986
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
43.884892086330936
0.0
0.0
0.0
4.892086330935252


In [24]:
data

Marca temporal    Carrera profesional cursada  \
0   2023-03-07 16:28:18.869               Ing. Electrónica   
1   2023-03-10 16:57:54.679               Ing. Electrónica   
2   2023-03-10 18:05:01.889               Ing. Electrónica   
3   2023-03-15 10:28:58.280               Ing. Electrónica   
4   2023-03-15 13:29:41.387                 Ing. Biomédica   
..                      ...                            ...   
690 2023-11-28 08:48:55.010               Ing. Electrónica   
691 2023-11-28 09:00:06.089               Ing. Electrónica   
692 2023-11-28 09:56:00.999               Ing. Electrónica   
693 2023-11-28 15:36:51.525               Ing. Electrónica   
694 2023-11-28 15:53:21.691  Ing. Sistemas Computacionales   

               Lugar dónde reside Titulado  Dominio del idioma Inglés  \
0            Hermosillo, Son, Mex       No                        0.8   
1     Hermosillo, Sonora, México        Si                        1.0   
2       Hermosillo,Sonora,Mexico        Si                        0.6   
3      Hermosillo Sonora, Mexico        Si                        0.8   
4    Guadalajara, Jalisco, Mexico       Si                        0.8   
..                            ...      ...                        ...   
690                    Hermosillo       Si                        0.6   
691      Hermosillo Sonora Mexico       No                        0.8   
692    Hermosillo, Sonora, Mexico       No                        0.4   
693            Hermosillo, Sonora       Si                        0.6   
694    Hermosillo, Sonora, Mexico       No                        0.8   

    Indica semestre y año de ingreso  Indica semestre y año de egreso  \
0                 2016-08-22 00:00:00             2020-12-18 00:00:00   
1                 2015-08-10 00:00:00             2019-12-02 00:00:00   
2                 2017-01-31 00:00:00             2020-06-20 00:00:00   
3                 2016-08-30 00:00:00             2021-01-20 00:00:00   
4                 2017-08-20 00:00:00             2022-01-10 00:00:00   
..                                ...                             ...   
690           Agosto - diciembre 2018              Enero - junio 2023   
691             Agosto Diciembre 2013           agosto Diciembre 2018   
692                       Agosto-2016                      JULIO-2019   
693           Agosto - Diciembre 2015              Enero - Junio 2020   
694           Agosto - Diciembre 2017              Enero - Junio 2022   

    ¿Tu egreso fue hace más de 2 años?  \
0                                   Si   
1                                   Si   
2                                   Si   
3                                   Si   
4                                   No   
..                                 ...   
690                                 No   
691                                 Si   
692                                 Si   
693                                 Si   
694                                 No   

    Eficiencia para realizar las actividades laborales, en relación con su formación académica  \
0                                            Eficiente                                           
1                                        Muy eficiente                                           
2                                            Eficiente                                           
3                                            Eficiente                                           
4                                        Muy eficiente                                           
..                                                 ...                                           
690                                          Eficiente                                           
691                                      Muy eficiente                                           
692                                      Muy eficiente                                           
693                          

In [25]:
path2Save = 'Segunda_Revision_Columnas_Limpiadas_Encuesta.xlsx'
data.to_excel(path2Save, index = False)